lib/Indexer/Adapter/Php/FileSearchIndex.php

```diff
@@ -56,7 +57,12 @@ public function search(Criteria $criteria): Generator
     public function write(Record $record): void
     {
         $this->open();
-        $info = [$record->recordType(), $record->identifier(), $record instanceof ClassRecord ? $record->type() : null];
+        $info = [
+            $record->recordType(),
+            $record->identifier(),
+            $record instanceof ClassRecord ? $record->type() : null,
+            $record instanceof ClassRecord ? $record->flags() : null,
```

Comment:
could generalize `flags` to the `Record` class I guess?

Reply:
`Record` is an interface. I am not sure if it needs `flags` method. Let's do it in a separate PR or please make a change here.

Reply:
or possible a `HasFlags` interface, but maybe premature it's fine.